rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+
+The viewing key adds another layer of transparency, allowing users to decrypt and review their transactions without compromising their authority over their assets. 
+While it provides a window into transaction history, the viewing key does not grant spending power, preserving the critical separation between visibility and control.
+
+The dual system of public and private addresses gives users the flexibility to navigate between open, transparent transactions and confidential, privacy-protected activities. 
+Public addresses allow users to engage in verifiable, public interactions, while private addresses enable compartmentalized, secure transactions that remain hidden 
+from the public eye. This dual-address framework ensures that users can seamlessly adapt to different privacy requirements, whether they are participating in public 
+dApps or conducting sensitive financial operations.
+
+Overall, Nescience’s cryptographic infrastructure is designed to empower users to engage confidently in both transparent and confidential activities. 
+By providing flexible, secure key management and address systems, Nescience ensures that users can fully participate in the blockchain ecosystem without 
+compromising their privacy or control. The architecture supports the nuanced needs of modern blockchain users, who require both the transparency of public 
+interactions and the security of private transactions, all while maintaining the integrity and confidentiality of their assets.
+
+
+## c) Trees in Nescience Architecture
+Trees in the Nescience architecture serve as verifiable databases essential for maintaining privacy and security. Different types of trees are used for various purposes:
+
+1. Global State Tree: The Global State Tree is a single, public tree that holds all public assets and storage information. It acts as a central repository for all 
+publicly accessible data on the blockchain. By organizing this data in a Merkle tree structure, the Global State Tree allows for efficient and secure verification of public information.
+
+2. Hashed UTXO Tree: The Hashed UTXO Tree is a public tree that contains hashes of all created UTXOs. When users wish to consume a UTXO, they provide a membership 
+proof to demonstrate that the UTXO exists within this tree. This process ensures that only valid and existing UTXOs can be spent, maintaining the integrity of transactions. 
+In fact, users generate membership proofs that verify the presence of specific UTXOs in the tree without revealing their actual data. The benefit here is that the Merkle 
+tree structure allows for quick and efficient verification of UTXO existence.
+
+3. UTXO Trees (Private States): Each user or smart contract (SC) has its private state stored in UTXO trees. These trees are kept as plaintext on the client’s 
+local system (off-chain), ensuring privacy as sensitive information remains confidential. The private state includes all UTXOs owned by the user or SC, and these 
```

Comment:
"offchain"
spell out SC as we didn't end up defining smart contract as an acronym